src/context/Post.ts

```diff
@@ -270,6 +274,7 @@ export class Data {
         downvote: number = 0,
         minRep = 0
     ) {
+        let metadata: Metadata = { id: postId ? postId : commentId }
```

Comment:
I guess here there's _some_ logic in the shared variable but I'm still not a fan. I'd rather have this ternary twice inside each of the scopes below.

Also this can be a `const`